### PR TITLE
Updated Benchmark and Fixed Code For Purity and Readable

### PR DESCRIPTION
**Pull request by Walker-00**
Since The benchmark is made at 2021, almost everything has changed not just for tealdeer almost every tldr clients has been changed so I made a re benchmarking and rewrite some code for readable purpose.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
-        toolchain: [stable, 1.64.0]
+        toolchain: [stable, 1.76.0]
```
Comment (niklasmohrin) [inline .github/workflows/ci.yml:17]: What is the motivation for this change? Your commit message suggests that you intended to update the version that is used for the release build, but we are already using the latest stable version for that. This number here is our Minimum Supported Rust Version (MSRV) and it should be kept as low as possible with the current code. Is there anything that we need from 1.76?

File: Cargo.toml [hunk 2]
```diff
@@ -66,3 +66,8 @@ strip = true
 opt-level = 3
 lto = true
 codegen-units = 1
+
+[profile.release.package."*"]
```
Comment (niklasmohrin) [inline Cargo.toml:70]: What does this do? Dont we only have one package in our repo anyways? Or does it apply to our dependencies? (Although I would expect that the other config block above this one covers everything)

File: src/main.rs [hunk 3]
```diff
@@ -200,7 +204,7 @@ fn init_log() {
 }
 
 #[cfg(not(feature = "logging"))]
-fn init_log() {}
+const fn init_log() {}
```
Comment (niklasmohrin) [inline src/main.rs:207]: This makes the signature incompatible with the other `init_log` (when logging is enabled). Please revert

File: src/main.rs [hunk 4]
```diff
@@ -209,7 +213,9 @@ fn get_languages(env_lang: Option<&str>, env_language: Option<&str>) -> Vec<Stri
     if env_lang.is_none() {
         return vec!["en".to_string()];
     }
-    let env_lang = env_lang.unwrap();
+
+    let env_lang = env_lang
+        .expect("Sorry It Shouldn't Be Happened We've checked and the env lang is not None!");
```
Comment (niklasmohrin) [inline src/main.rs:218]: I think the version with `unwrap` is better here. If we really want to get rid of unwrap, we should use a let-else statement (which was not stable when this code was written)

File: src/main.rs [hunk 5]
```diff
@@ -26,9 +28,10 @@
         feature = "native-tls"
     )),
 ))]
-compile_error!(
+/*compile_error!(
     "exactly one of the features \"native-roots\", \"webpki-roots\" or \"native-tls\" must be enabled"
-);
+);*/
+print!("hi");
```
Comment (niklasmohrin) [inline src/main.rs:34]: Please revert

**Review by niklasmohrin — CHANGES_REQUESTED** [review]
Haven't looked into the outfieldr stuff yet, although I think that if we dont get it compile we should just remove it. 

Was the impact of all the `const` significant?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Walker-00) [reply .github/workflows/ci.yml:17]: Yea, my mistake when I update the dep crates `regex-syntax v0.8.2` need MSRV of 1.65.0, Sorry I'll fix it.

File: src/main.rs (hunk 5 above)
Comment (Walker-00) [reply src/main.rs:34]: Yea I forget to delete that and expect code, I used them to ignore the errors msg from clippy and lsp, Sorry again I'll fix it.

File: Cargo.toml (hunk 2 above)
Comment (Walker-00) [reply Cargo.toml:70]: Think it would cool to add that :3

**Comment by niklasmohrin** [comment]
@Walker-00 I take it you do not want to continue working on this (which is totally fine)? Otherwise, I would suggest that we split the code changes and the benchmark changes into two separate PRs to make reviewing easier

**Comment by Walker-00** [comment]
> @Walker-00 I take it you do not want to continue working on this (which is totally fine)? Otherwise, I would suggest that we split the code changes and the benchmark changes into two separate PRs to make reviewing easier

Nah bro, I really enjoy contributing to this pj. But I'm a bit busy lately with my startup. I'll keep contributing when I'm free.

**Comment by niklasmohrin** [comment]
Okay, no worries

**Comment by Walker-00** [comment]
yea bro, so should I reopen pull req

**Comment by niklasmohrin** [comment]
Pleaae leave it closed until you want to resume working on it :)

**Comment by Walker-00** [comment]
oki bro
